airbyte-cdk/bulk/core/base/src/testFixtures/kotlin/io/airbyte/cdk/command/CliRunner.kt

```diff
@@ -33,54 +40,73 @@ data object CliRunner {
         config: ConfigurationJsonObjectBase? = null,
         catalog: ConfiguredAirbyteCatalog? = null,
         state: List<AirbyteStateMessage>? = null,
-    ): BufferingOutputConsumer =
+    ): BufferingOutputConsumer {
+        val out = TestOutputStream()
         runConnector(op, config, catalog, state) { args: Array<String> ->
-            AirbyteSourceRunner(args)
+            AirbyteSourceRunner(args, out.beanDefinition)
         }
+        return out.results
+    }
 
     /** Same as [runSource] but for destinations. */
     fun runDestination(
         op: String,
         config: ConfigurationJsonObjectBase? = null,
         catalog: ConfiguredAirbyteCatalog? = null,
         state: List<AirbyteStateMessage>? = null,
-    ): BufferingOutputConsumer =
+        inputStream: InputStream,
+    ): BufferingOutputConsumer {
+        val inputBeanDefinition: RuntimeBeanDefinition<InputStream> =
+            RuntimeBeanDefinition.builder(InputStream::class.java) { inputStream }
+                .singleton(true)
```

Comment:
nit: if you don't want to need the `Requires(notEnv=TEST)` in InputConsumer, you can do `.replaces(InputStream::class.java)`

(I have no opinion on whether that's preferable :shrug:)

Reply:
There were concurrent changes which got merged which annotated InputStreamFactory as Secondary so I don't need to do anything now.